event-sourcing/event-sourcing-core/src/main/scala/org/apache/james/eventsourcing/CommandDispatcher.scala

```diff
@@ -45,9 +50,17 @@ object CommandDispatcher {
 class CommandDispatcher @Inject()(eventBus: EventBus, handlers: Set[CommandHandler[_ <: Command]]) {
   Preconditions.checkArgument(hasOnlyOneHandlerByCommand(handlers), CommandDispatcher.ONLY_ONE_HANDLER_PRECONDITION)
 
-  def dispatch(c: Command): Unit = {
-    trySeveralTimes(() => tryDispatch(c))
-      .getOrElse(() => throw CommandDispatcher.TooManyRetries(c, CommandDispatcher.MAX_RETRY))
+  def dispatch(c: Command): Publisher[Void] = {
+    tryDispatch(c)
+      .retry(CommandDispatcher.MAX_RETRY, {
+        case _: EventStoreFailedException => true
+        case _ => false
+      })
+      .onErrorMap({
+        case _: EventStoreFailedException => CommandDispatcher.TooManyRetries(c, CommandDispatcher.MAX_RETRY)
+        case error => error
+      })
+      .`then`()
```

Comment:
What is this `then` for? `tryDispatch` is already returning `SMono[Void]` right?